src/main/java/codechicken/nei/GuiNEIButton.java

```diff
@@ -40,25 +39,27 @@ public void drawButton(Minecraft minecraft, int i, int j) {
                 46 + k * 20 + 20 - height / 2,
                 width / 2,
                 height / 2); // bottom right
-        mouseDragged(minecraft, i, j);
+        mouseDragged(minecraft, x, y);
 
-        if (!enabled) drawCenteredString(
-                fontrenderer,
-                displayString,
-                xPosition + width / 2,
-                yPosition + (height - 8) / 2,
-                0xffa0a0a0);
-        else if (flag) drawCenteredString(
-                fontrenderer,
-                displayString,
-                xPosition + width / 2,
-                yPosition + (height - 8) / 2,
-                0xffffa0);
-        else drawCenteredString(
-                fontrenderer,
-                displayString,
-                xPosition + width / 2,
-                yPosition + (height - 8) / 2,
-                0xe0e0e0);
+        drawContent(minecraft, x, y, mouseOver);
+    }
+
+    protected int getTextColour(boolean mouseOver) {
```

Comment:
do you really need to extract this to protected methods ? are you going to override this anywhere ?

I'd prefer that you just leave it inside one method like before and do

```JAVA
        final int color;
        if (!enabled) {
            color = 0xffa0a0a0;
        } else if (mouseOver) {
            color = 0xffffa0;
        } else {
           color = 0xe0e0e0;
        }
```

Reply:
I need this for another PR I'm working on.

Reply:
okay